crates/ra_tt/src/lib.rs

```diff
@@ -149,3 +149,15 @@ impl fmt::Display for Punct {
         fmt::Display::fmt(&self.char, f)
     }
 }
+
+impl Subtree {
+    /// Count the number of tokens recursively
+    pub fn count(&self) -> usize {
+        self.token_trees.iter().fold(self.token_trees.len(), |acc, c| {
```

Comment:
It could be expressed more clearly as a combination of `.map` and `.sum`. r=me with this change